Riot/Modules/MatrixKit/Models/Room/MXKRoomDataSource.m

```diff
@@ -1923,6 +1924,67 @@ - (BOOL)canReplyToEventWithId:(NSString*)eventIdToReply
     return [self.room canReplyToEvent:eventToReply];
 }
 
+// Check filesize before sending: check file size before sending
+- (BOOL)_isFilesizeOkToBeSent:(NSUInteger)filesize
+{
+    // Check maxUploadSize accepted by the home server before trying to upload.
+    NSUInteger maxUploadFileSize = self.mxSession.maxUploadSize;
+    if (filesize > maxUploadFileSize)
+    {
+        return NO;
+    }
+    else
+    {
+        return YES;
+    }
+}
+
+- (BOOL)isFilesizeOkToBeSentForData:(NSData *)fileData
+{
+    return [self _isFilesizeOkToBeSent:fileData.length];
+}
+
+- (BOOL)isFilesizeOkToBeSentForLocalFileUrl:(NSURL *)localFileUrl
+{
+    NSDictionary *fileAttributes = [NSFileManager.defaultManager attributesOfItemAtPath:localFileUrl.path error:nil];
+    if (fileAttributes)
+    {
+        return [self _isFilesizeOkToBeSent:fileAttributes.fileSize];
+    }
+    else
+    {
+        return NO;
+    }
+}
+
+- (BOOL)isFilesizeOkToBeSentForLocalAVAsset:(AVAsset *)asset
+{
+    // Check if asset points to a local file
+    if( ![asset isKindOfClass:AVURLAsset.class] )
```

Comment:
Our objective-c style should look like so:

```suggestion
    if (![asset isKindOfClass:AVURLAsset.class])
```

There's a few of these in here.